api/v1beta1/types.go

```diff
@@ -799,6 +809,14 @@ type AzureBastion struct {
 	Subnet SubnetSpec `json:"subnet,omitempty"`
 	// +optional
 	PublicIP PublicIPSpec `json:"publicIP,omitempty"`
+	// BastionHostSkuName configures the tier of the Azure Bastion Host. Can be either Basic or Standard. Defaults to Basic.
+	// +kubebuilder:validation:Enum=Basic;Standard
+	// +optional
+	Sku BastionHostSkuName `json:"sku,omitempty"`
+	// EnableTunneling enables the native client support feature for the Azure Bastion Host. Defaults to false.
```

Comment:
follow up question: are any of these new fields immutable? if so we should add webhook validation to prevent users from getting in a bad state

Reply:
I'm not entirely sure. I think the SKU is able to be upgraded from Basic to Standard, but not the other way around. I'll test and see if the upgrade can be done through the spec, and add validation to prevent the downgrade.

EnableTunneling is also only supported via the Standard SKU, so we won't be able to default to it. I will add more webhook validation here, I shouldn't have missed this!

Reply:
Update: Looks like there is already [validation](https://github.com/willie-yao/cluster-api-provider-azure/blob/364f124427e3444481942bf53a98eb6d840c00e5/api/v1beta1/azurecluster_webhook.go#L117) that prevents any fields from the bastion to be changed at all.